### PR TITLE
Fix macro redefinition warning from armclang

### PR DESCRIPTION
**Pull request by daverodgman**
## Description

Avoid redefining the MBEDTLS_BSWAPxx macros for compilers which support multiple options, such as armclang.

Fixes some of #7186 

## Gatekeeper checklist

- [x] **changelog** not required (not previously released)
- [x] **backport** not required - not in 2.28
- [x] **tests** not required

### CODE REVIEW

**Comment by daverodgman** [comment]
While I'm here, also fix some annoying whitespace mis-alignments introduced by the code-style change

File: library/alignment.h [hunk 1]
```diff
@@ -155,13 +155,13 @@ inline void mbedtls_put_unaligned_uint64(void *p, uint64_t x)
  * Detect Clang built-in byteswap routines
  */
 #if defined(__clang__) && defined(__has_builtin)
```
Comment (tom-cosgrove-arm) [inline library/alignment.h:157]: Ah, we should actually check for clang first, then (in an else) for gcc, since clang also defines `__GNUC__`

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Review by paul-elliott-arm — APPROVED** [review]
LGTM

**Pull request merged**